lib/spack/spack/cmd/fetch.py

```diff
@@ -33,6 +33,9 @@ def fetch(parser, args):
     if args.no_checksum:
         spack.config.set('config:checksum', False, scope='command_line')
 
+    if args.deprecated:
+        spack.config.set('config:deprecated', True, scope='command_line')
```

Comment:
This lets me know there is already a mapping for command-line vs yaml options in Spack, which is cool because I would like config options to actually be fully fungible with command-line and environment variables [as in pants](https://v1.pantsbuild.org/options.html). I'll open an issue or PR about that separately.